Packet++/header/FtpLayer.h

```diff
@@ -0,0 +1,458 @@
+			/// Store file uniquely.
+			STOU = ('S') | ('T' << 8) | ('O' << 16) | ('U' << 24),
+			/// Set file transfer structure.
+			STRU = ('S') | ('T' << 8) | ('R' << 16) | ('U' << 24),
+			/// Return system type.
+			SYST = ('S') | ('Y' << 8) | ('S' << 16) | ('T' << 24),
+			/// Get a thumbnail of a remote image file
+			THMB = ('T') | ('H' << 8) | ('M' << 16) | ('B' << 24),
+			/// Sets the transfer mode (ASCII/Binary).
+			TYPE = ('T') | ('Y' << 8) | ('P' << 16) | ('E' << 24),
+			/// Authentication username.
+			USER = ('U') | ('S' << 8) | ('E' << 16) | ('R' << 24),
+			/// Change to the parent of the current working directory
+			XCUP = ('X') | ('C' << 8) | ('U' << 16) | ('P' << 24),
+			/// Make a directory
+			XMKD = ('X') | ('M' << 8) | ('K' << 16) | ('D' << 24),
+			/// Print the current working directory
+			XPWD = ('X') | ('P' << 8) | ('W' << 16) | ('D' << 24),
+			///
+			XRCP = ('X') | ('R' << 8) | ('C' << 16) | ('P' << 24),
+			/// Remove the directory
+			XRMD = ('X') | ('R' << 8) | ('M' << 16) | ('D' << 24),
+			///
+			XRSQ = ('X') | ('R' << 8) | ('S' << 16) | ('Q' << 24),
+			/// Send, mail if cannot
+			XSEM = ('X') | ('S' << 8) | ('E' << 16) | ('M' << 24),
+			/// Send to terminal
+			XSEN = ('X') | ('S' << 8) | ('E' << 16) | ('N' << 24)
+		};
+
+		/** A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		FtpRequestLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet) : FtpLayer(data, dataLen, prevLayer, packet) { m_Protocol = FTP; };
+
+		/**
+		 * Empty c'tor
+		 */
+		FtpRequestLayer();
+
+		/**
+		 * Set the command of request message
+		 * @param[in] code Value to set command
+		 */
+		void setCommand(FtpCommand code);
+
+		/**
+		 * Get the command of request message
+		 * @return FtpCommand Value of the command
+		 */
+		FtpCommand getCommand() const;
+
+		/**
+		 * Get the command of request message as string
+		 * @return std::string Value of the command as string
+		 */
+		std::string getCommandString() const;
+
+		/**
+		 * Set the command argument of request message
+		 * @param[in] value Value to set command argument
+		 */
+		void setCommandOption(std::string value);
+
+		/**
+		 * Get the command argument of request message
+		 * @param[in] removeEscapeCharacters Whether non-alphanumerical characters should be removed or not
+		 * @return std::string Value of command argument
+		 */
+		std::string getCommandOption(bool removeEscapeCharacters = true) const;
+
+		/**
+		 * Convert the command info to readable string
+		 * @param[in] code Command code to convert
+		 * @return std::string Returns the command info as readable string
+		 */
+		static std::string getCommandInfoAsString(FtpCommand code);
```

Comment:
We can rename to just `getCommandInfo()`, the user can see it returns a `std::string` 😄